### DISCUSSION TITLE
Expand model fields only without sensitive data

### DISCUSSION THREAD

**Discussion by tarikrital** [swellstores/community#496] [Help] [answered]
Welcome, I was able to get the Vendor (Customer) using `expand` from the previous discussion [closed](https://github.com/orgs/swellstores/discussions/495).

Now I want to select only 1 or 2 fields i.e `first_name` and `name`.

Is that possible by using `expand: ["vendor.first_name"]`?

```
....
const product = await swell.get(`/products/${id}`, {
      id,
      expand: ["vendor"]
...
```

**Comment by ericingram** [comment, marked as answer]
@wpplumber good question. Did you try it? Can't remember if that works but it is similar to semantics for nested links

If not, you can use an `include` statement, for example:

```
const product = await swell.get(`/products/${id}`, {
  id,
  include: {
    vendor: {
      url: '/products/{id}/vendor', // note: this would be more efficient if you use the absolute vendor endpoint here instead of the nested path starting with /products/{id}
      data: {
        fields: 'first_name'
      }
    }
  }
})
```

**Reply by tarikrital** [reply]
>Did you try it?

I does't work but your suggestion works fine.

>url: '/products/{id}/vendor', // note: this would be more efficient if you use the absolute vendor endpoint here instead of the nested path starting with /products/{id}

Is this something mentioned on the doc or a tip?

**Reply by ericingram** [reply]
Just a tip. Each nested link in a URL causes an additional DB request, thus it's more efficient to go directly to the target record when possible.

**Reply by tarikrital** [reply]
Thank you, Eric.
